tests/snapshot.test.ts

```diff
@@ -60,3 +60,10 @@ it('should not cause proxy-compare to copy', async () => {
   const cmp = createProxy(snap1, new WeakMap())
   expect(getUntracked(cmp)).toBe(snap1)
 })
+
+it('should create a new proxy from a snapshot', async () => {
+  const state = proxy({ c: 0 })
+  const snap1 = snapshot(state)
+  const state2 = proxy(snap1)
+  expect(state2.c).toBe(0)
```

Comment:
@dai-shi with this approach, this assertion fails:

```
expect(state2).toBe(state);
```

Wdyt of teaching `proxy` to realize "I'm being passed a snapshot" and then using some sort of `getOriginalObject` / `getUntracked` approach to just recover the existing proxy?

It seems like it'd be nice to get back the same store identity.

Or, alternatively, just throw an error and say "you passed a snapshot to proxy...don't do that".

Reply:
> Or, alternatively, just throw an error and say "you passed a snapshot to proxy...don't do that".

Wouldn't the be a breaking change?

Reply:
Yes, I'm just thinking if @dai-shi thinks the "proxying a snapshot" is enough of a unintended use case / probably not what the user really wanted to do, it'd be better to just throw and guide them to a better pattern, i.e. specifically passing the store around directly (which I believe is the recommended approach).

(Or, maybe we do the 1st option, which is that "proxying the snapshot" does become the canonical way of getting "back to the store" from the snapshot, which again AFAIU is unintended/accidental behavior today.)

Reply:
Maybe this the misguided. I think we should support creating a new proxy from a frozen (incl. non-configurable&non-writable) object. And, a snap is just an example of it. And we support it.

It's definitely not about recovering the original proxy.

Reply:
> I think we should support creating a new proxy from a frozen

Ah sure; is that what @octet-stream wants to do? I might have misread the thread, but I thought the idea was to get back to the original store.

I guess Nick's comment "so that I can use it (the snapshot) to create a new proxy" does explicitly ask for going from snapshot --> "new store", but the intervening comments made me jump to the conclusion Nick was running into the "how do I get a store in a child that was passed a snapshot" wrinkle.

And not literally "I want a completely separate / 2nd additional store that is 100% detached from the first store".

But I could have made the wrong assumption.

Reply:
From @octet-stream 's example, he almost certainly wants the original store; he's using a `useSnapshot` against a root list of notes:

```
  const {items} = useNotesStateSnapshot()

  return (
    <ul>
      {items.map((note) => (
        <li key={note.id}>
          <NoteStateContextProvider data={note}>
            <NoteCard />
          </NoteStateContextProvider>
        </li>
      ))}
    </ul>
```

Passing each note (a compare proxy) into a child context and doing `proxy(data)`:

```
  const StateContextProvider: FC<ProviderProps<T>> = ({data, children}) => {
    // Use `useMemo` instead of `useRef` with `valtio`, so that we can reload page state if the `data` is changed
    const state = useMemo(() => proxy(data), [data])
```

Which I just really have to assume the desire is "get me the note[0] model from the original store, so I can mutate it in my child if I want":

This is why I think calling `proxy(snapshotFromSnapshot)` or `proxy(snapshotProxyFromUseSnapshot)` should either:

a) throw a "this is probably not what you meant to do" un-intended use error, or 
b) return the original underlying store that creates the `snapshot` snapshot or `useSnapshot 

Instead of the current behavior which, AFAIU creates a 2nd store that is 100% disconnected (different data, different subscribers) from the 1st store and just surely is not what the user wanted.

Reply:
> he almost certainly wants the original store

It might be the case, and if so, we could do it right? https://github.com/pmndrs/valtio/discussions/657#discussioncomment-4878032

Reply:
Yes. @octet-stream can you confirm if @dai-shi 's linked comment is what you're trying to do?

Fwiw @dai-shi this is what the docs PR I opened tries to clarify for users of useSnapshot; it has your example as-is, explaining that they should pass the store if they want to mutate.

I don't think I included a disclaimer that if they proxy a snapshot, it will make a completely separate store, but that's probably a good idea.